Moya.swift

```diff
@@ -99,6 +99,25 @@ public protocol MoyaTarget : MoyaPath {
     var sampleData: NSData { get }
 }
 
+/// Protocol to define the opaque type returned from a request
+public protocol Cancelable {
+    func cancel()
+}
+
+/// Internal token that can be used to cancel requests
+struct CancelableToken : Cancelable {
+    
+    let cancelAction: () -> ()
+    
+    init(cancelAction: () -> ()) {
+        self.cancelAction = cancelAction
+    }
+    
+    func cancel() {
```

Comment:
Do we care about cancelling things more than once? Probably not. I'd rather not expose state anyway :)